build.rs

```diff
@@ -14,24 +14,35 @@ fn build_syntax(dir: &str, out: &str) {
     dump_to_file(&ss, Path::new(&out_dir).join(out)).unwrap();
 }
 
-fn feature_status(feature: &str) -> String {
-    if env::var_os(format!(
+fn feature_enabled(feature: &str) -> bool {
+    env::var_os(format!(
         "CARGO_FEATURE_{}",
         feature.to_uppercase().replace('-', "_")
     ))
     .is_some()
-    {
+}
+
+fn feature_status(feature: &str) -> String {
+    if feature_enabled(feature) {
         format!("+{}", feature)
     } else {
         format!("-{}", feature)
     }
 }
 
 fn features() -> String {
-    feature_status("native-tls")
+    format!(
+        "{} {}",
+        &feature_status("native-tls"),
+        &feature_status("rustls")
+    )
 }
 
 fn main() {
+    if !feature_enabled("native-tls") && !feature_enabled("rustls") {
+        panic!("Either native-tls or rustls feature must be enabled!");
+    }
```

Comment:
Maybe this should be a compile-time error? Not sure what the best practices for that are.

Reply:
Maybe we could use the [`compile_error!()`](https://doc.rust-lang.org/std/macro.compile_error.html) macro?

#### `panic!()` inside build.rs
```
$ cargo run --no-default-features -- --help
   Compiling xh v0.16.1 (/home/ducaale/xh)
   Compiling reqwest v0.11.10
error: failed to run custom build command for `xh v0.16.1 (/home/ducaale/xh)`

Caused by:
  process didn't exit successfully: `/home/ducaale/xh/target/debug/build/xh-92f620f356085f57/build-script-build` (exit status: 101)
  --- stderr
  thread 'main' panicked at 'Either native-tls or rustls feature must be enabled!', build.rs:43:9
  note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

#### `compile_error!()` inside build.rs
```
$ cargo run --no-default-features -- --help
   Compiling xh v0.16.1 (/home/ducaale/xh)
error: Either native-tls or rustls feature must be enabled!
  --> build.rs:43:5
   |
43 |     compile_error!("Either native-tls or rustls feature must be enabled!");
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error: could not compile `xh` due to previous error
```

Reply:
> Not sure what the best practices for that are.

Me neither.

> Maybe we could use the `compile_error!()` macro?

Yeah, this looks like the right macro for this use case! I’ve updated the PR.